lib/fog/ovirt/requests/compute/v4/update_vm.rb

```diff
@@ -0,0 +1,48 @@
+module Fog
+  module Compute
+    class Ovirt
+      class V4
+        class Real
+          # rubocop:disable Metrics/AbcSize
+          def update_vm(attrs)
+            attrs = attrs.dup
+            vm_service = client.system_service.vms_service.vm_service(attrs[:id])
+
+            if attrs[:cores].present?
+              cpu_topology = OvirtSDK4::CpuTopology.new(:cores => attrs[:cores], :sockets => "1")
+              attrs[:cpu] = OvirtSDK4::Cpu.new(:topology => cpu_topology)
+            end
+            attrs[:cluster] = if attrs[:cluster].present? ? OvirtSDK4::Cluster.new(:id => attrs[:cluster]) : nil
+            attrs[:template] = if attrs[:template].present?
+                                 OvirtSDK4::Template.new(:id => attrs[:template])
+                               else
+                                 blank_template
+                               end
+
+            attrs[:large_icon] = if attrs[:large_icon].present? ? OvirtSDK4::Icon.new(:id => attrs[:large_icon]) : nil
+            attrs[:small_icon] = if attrs[:small_icon].present? ? OvirtSDK4::Icon.new(:id => attrs[:small_icon]) : nil
+            attrs[:cpu_profile] = if attrs[:cpu_profile].present? ? OvirtSDK4::CpuProfile.new(:id => attrs[:cpu_profile]) : nil
+            attrs[:host] = if attrs[:host].present? ? OvirtSDK4::Host.new(:id => attrs[:host]) : nil
+            attrs[:quota] = if attrs[:quota].present? ? OvirtSDK4::Quota.new(:id => attrs[:quota]) : nil
+            attrs[:high_availability] = OvirtSDK4::HighAvailability.new(:enabled => attrs[:ha]) unless attrs[:ha].nil?
+            attrs[:instance_type] = if attrs[:instance_type].present? ? OvirtSDK4::InstanceType.new(:id => attrs[:instance_type]) : nil
```

Comment:
@ShimShtein after this failed testing I saw why I had to use the `?`, `"".present?` returns false and if the empty string is sent instead of a SDK object or nil the SDK writers will fail.

Reply:
it chimes in with a later comment - we are sending `attrs` as is to `vm_service`. I think we should remove them from the original hash.